pkg/loadbalancer/redirectpolicy/redirectpolicy.go

```diff
@@ -218,6 +211,25 @@ func getSanitizedLocalRedirectPolicy(name, namespace string, uid types.UID, spec
 		if err != nil {
 			return nil, fmt.Errorf("invalid address matcher IP %v: %w", addrMatcher.IP, err)
 		}
+
+		if !cfg.addressAllowed(addrCluster.Addr()) {
+			log.Warn("Ignoring IP in AddressMatcher disallowed by --"+AddressMatcherCIDRsName,
+				logfields.Address, addrMatcher.IP,
+				logfields.CIDRs, cfg.AddressMatcherCIDRs,
+				logfields.K8sNamespace, namespace,
+				logfields.Name, name)
+
+			// Return an "empty" policy so that any prior policy will be flushed out.
```

Comment:
Why do we clear out prior state -- is it not enough to just reject the current policy that specifies an address disallowed by `AddressMatcherCIDRS`?

Reply:
Yeah good point. It would retain the old policy, but that's likely what most users would expect anyway. Will change it to reject instead.

Reply:
I don't quite follow -- if a frontend address specified in an LRP is disallowed, then we reject the policy. The LRP manager plumbs a redirect service entry for a frontend -> backend(s) as derived from an LRP.  What prior state are you referring to -- it is some state stored in stateDB or the BPF LB maps?